src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -582,6 +598,33 @@ def generate_request_payload(configuration, location, public_key, tags, kubernet
         distribution=kubernetes_distro,
         infrastructure=kubernetes_infra
     )
+
+    if enable_private_link:
+        private_link_state = "Enabled" if enable_private_link.lower() == "true" else "Disabled"
```

Comment:
This condition seems redundant here as line#602 already checks the same conditon.

Reply:
It's not redundant. Even if the --enable-private-link is passed as 'false', the if statement will return true. The if statement will return false in case user does not pass the parameter --enable-private-link

Reply:
Should we check `if not enable_private_link.lower() == "true"` rather than empty if clause?